powermon/ports/__init__.py

```diff
@@ -1,45 +1,37 @@
+from mppsolar.protocols import get_protocol
+
 import importlib
-import logging
-from enum import Enum, auto
 
+from powermon.ports.abstractport import PortType
+from powermon.ports.serialport import SerialPort
+from powermon.ports.usbport import USBPort
 
-class PortType(Enum):
-    UNKNOWN = auto()
-    TEST = auto()
-    USB = auto()
-    ESP32 = auto()
-    SERIAL = auto()
-    JKBLE = auto()
-    MQTT = auto()
-    VSERIAL = auto()
-    DALYSERIAL = auto()
-    BLE = auto()
+def getPortFromConfig(port_config):
 
+    # get protocol handler
+    protocol = get_protocol(protocol=port_config["protocol"])
+    #log.debug(f"protocol: {protocol}")
 
-log = logging.getLogger("ports")
+    portType = port_config["type"]
+    #Only port type is mandatory
+    #TODO: move configuration parsing into port class constructors
+    portPath = port_config.get("path", None)
+    portBaud = port_config.get("baud", None)
+    # return None if port type is not defined
+    if portType is None:
+        return None
 
+    portObject = None
 
-def get_port(config):
-    log.info(f"Geting port for config '{config}'")
-    porttype = config.get("type", None)
+    if portType == PortType.SERIAL:
+        portObject = SerialPort(portPath, portBaud, protocol)
+    elif portType == PortType.USB:
+        portObject = USBPort(portPath, protocol)
 
-    # return None if port type is not defined
-    if porttype is None:
-        return None
+    #Pattern for port types that cause problems when imported
```

Comment:
think we should do this for all ports - dont even need pyserial as a forced dependancy then